docs/interfaces/auth-service.txt

```diff
@@ -49,7 +77,7 @@ Login (username: string, password: string): (Token, Cookie);
 *
 * @throws InvalidCredentials
 */
-Authenticate (token: Token, cookie: Cookie): {userId: number; sessionId: string;};
+"/authenticate" (Headers: {'authentication': Token, cookie: refreshId=Cookie}): Response<T = LoginInformation>;
```

Comment:
Just keep it simple and add `authentication` and `set-cookie` in the response object as optional keys. Instead of using too much generics, maybe create a response type with and without headers.